apis/v1/gateway_types.go

```diff
@@ -654,6 +654,17 @@ type GatewayInfrastructure struct {
 	// +optional
 	// +kubebuilder:validation:MaxProperties=8
 	Annotations map[AnnotationKey]AnnotationValue `json:"annotations,omitempty"`
+
+	// ParametersRef is a reference to a resource that contains the configuration
+	// parameters corresponding to the Gateway. This is optional if the
+	// controller does not require any additional configuration.
+	//
+	// This follows the same semantics as GatewayClass's `parametersRef`, but on a per-Gateway basis
+	//
+	// Support: Implementation-specific
+	//
+	// +optional
+	ParametersRef *ParametersReference `json:"parametersRef,omitempty"`
```

Comment:
This allows for cross-namespace references, I'm not sure we want to allow for that. The [corresponding GEP](https://gateway-api.sigs.k8s.io/geps/gep-1867/) doesn't make it obvious if those are needed, do you mind updating the GEP to clarify the goals here?

Reply:
I made a new `LocalParametersReference` to force it to be same namespace